diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstancesResponse.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstancesResponse.java
index 2b9bd6a..66d0a57 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstancesResponse.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstancesResponse.java
@@ -29,12 +29,15 @@ public class ProcessInstancesResponse implements Serializable {
   String businessKey;
   String processDefinitionId;
   String startTime;
+  String startUserId;
+  String superProcessInstanceId;
   
   public ProcessInstancesResponse(HistoricProcessInstance processInstance) {
     this.setId(processInstance.getId());
     this.setBusinessKey(processInstance.getBusinessKey());
     this.setStartTime(RequestUtil.dateToString(processInstance.getStartTime()));
     this.setProcessDefinitionId(processInstance.getProcessDefinitionId());
+    this.setStartUserId(processInstance.getStartUserId());
   }
 
   public String getId() {
@@ -68,4 +71,20 @@ public class ProcessInstancesResponse implements Serializable {
   public void setProcessDefinitionId(String processDefinitionId) {
     this.processDefinitionId = processDefinitionId;
   }
+  
+  public String getStartUserId() {
+    return startUserId;
+  }
+
+  public void setStartUserId(String startUserId) {
+    this.startUserId = startUserId;
+  }
+
+  public String getSuperProcessInstanceId() {
+    return this.superProcessInstanceId;  
+  }
+
+  public void setSuperProcessInstanceId(String superProcessInstanceId) {
+    this.superProcessInstanceId = superProcessInstanceId;  
+  }
 }
