diff --git /dev/null b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/StateResponse.java
new file mode 100644
index 0000000..a1757a3
--- /dev/null
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/StateResponse.java
@@ -0,0 +1,14 @@
+package org.activiti.rest.api.identity;
+
+public class StateResponse {
+  private boolean success;
+
+  public boolean isSuccess() {
+    return success;
+  }
+
+  public StateResponse setSuccess(boolean success) {
+    this.success = success;
+    return this;
+  }
+}
diff --git /dev/null b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/UserInfo.java
new file mode 100644
index 0000000..99e4bf5
--- /dev/null
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/UserInfo.java
@@ -0,0 +1,51 @@
+package org.activiti.rest.api.identity;
+
+public class UserInfo {
+  
+  private String userId;
+  private String password;
+  private String firstName;
+  private String lastName;
+  private String email;
+
+  public String getUserId() {
+    return userId;
+  }
+
+  public void setUserId(String userId) {
+    this.userId = userId;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  public void setPassword(String password) {
+    this.password = password;
+  }
+
+  public String getFirstName() {
+    return firstName;
+  }
+
+  public void setFirstName(String firstName) {
+    this.firstName = firstName;
+  }
+
+  public String getLastName() {
+    return lastName;
+  }
+
+  public void setLastName(String lastName) {
+    this.lastName = lastName;
+  }
+
+  public String getEmail() {
+    return email;
+  }
+
+  public void setEmail(String email) {
+    this.email = email;
+  }
+
+}
diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/UserResource.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/UserResource.java
index 4787583..7dc004b 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/UserResource.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/identity/UserResource.java
@@ -14,16 +14,22 @@
 package org.activiti.rest.api.identity;
 
 import org.activiti.engine.ActivitiException;
+import org.activiti.engine.IdentityService;
 import org.activiti.engine.identity.User;
 import org.activiti.rest.api.ActivitiUtil;
 import org.activiti.rest.api.SecuredResource;
 import org.restlet.resource.Get;
+import org.restlet.resource.Put;
 
 /**
  * @author Tijs Rademakers
  */
 public class UserResource extends SecuredResource {
   
+  public UserResource(){
+    super();
+  }
+
   @Get
   public UserResponse getUser() {
     if(authenticate() == false) return null;
@@ -36,5 +42,29 @@ public class UserResource extends SecuredResource {
     UserResponse response = new UserResponse(user);
     return response;
   }
-
+  
+  @Put()
+  public StateResponse createUser(UserInfo userInfo){
+    if(authenticate() == false) return null;
+    
+    IdentityService identityService = ActivitiUtil.getIdentityService();
+    
+    String userId = userInfo.getUserId();
+    
+    if( userId == null) {
+      throw new ActivitiException("No user id supplied");
+    }
+    
+    if (identityService.createUserQuery().userId(userId).count() == 0) {
+      User user = identityService.newUser(userId);
+      user.setFirstName(userInfo.getFirstName());
+      user.setLastName(userInfo.getLastName());
+      user.setPassword(userInfo.getPassword());
+      user.setEmail(userInfo.getEmail());
+      identityService.saveUser(user);
+    }else{
+      throw new ActivitiException("user id must be unique");
+    }
+    return new StateResponse().setSuccess(true);
+  }
 }
diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java
index 2f02ac1..4cd8c28 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java
@@ -103,6 +103,7 @@ public class ActivitiRestApplication extends Application {
     
     router.attach("/login", LoginResource.class);
     
+    router.attach("/user", UserResource.class);
     router.attach("/user/{userId}", UserResource.class);
     router.attach("/user/{userId}/groups", UserGroupsResource.class);
     router.attach("/user/{userId}/picture", UserPictureResource.class);
