Index: modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstanceResource.java
===================================================================
--- modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstanceResource.java	(revision 3030)
+++ modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessInstanceResource.java	(working copy)
@@ -18,12 +18,16 @@
 import java.util.Map;
 
 import org.activiti.engine.ActivitiException;
 import org.activiti.engine.runtime.ProcessInstance;
 import org.activiti.rest.api.ActivitiUtil;
 import org.activiti.rest.api.SecuredResource;
 import org.codehaus.jackson.JsonNode;
 import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ArrayNode;
+import org.codehaus.jackson.node.ObjectNode;
 import org.restlet.representation.Representation;
+import org.restlet.resource.Get;
 import org.restlet.resource.Post;
 
 /**
@@ -73,4 +77,29 @@
       throw new ActivitiException("Failed to retrieve the process definition parameters", e);
     }
   }
+
+	@Get
+	public ObjectNode getVariables() {
+		if (authenticate() == false)
+			return null;
+		String processInstanceId = (String) getRequest().getAttributes().get("processInstanceId");
+		
+		Map<String, Object> variables = ActivitiUtil.getRuntimeService().getVariables(processInstanceId);
+
+		ObjectNode responseJSON = new ObjectMapper().createObjectNode();
+
+		ArrayNode propertiesJSON = new ObjectMapper().createArrayNode();
+
+		for (Iterator<String> it = variables.keySet().iterator(); it.hasNext();){
+			String key = it.next();
+			String value = String.valueOf(variables.get(key));
+			
+			ObjectNode propertyJSON = new ObjectMapper().createObjectNode();
+			propertyJSON.put(key,value);
+			propertiesJSON.add(propertyJSON);
+		}
+		responseJSON.put("data", propertiesJSON);
+
+		return responseJSON;
+	}
 }
Index: modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java
===================================================================
--- modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java	(revision 3030)
+++ modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/application/ActivitiRestApplication.java	(working copy)
@@ -102,7 +102,10 @@
     
     router.attach("/process-definitions", ProcessDefinitionsResource.class);
     router.attach("/process-instances", ProcessInstancesResource.class);
+    
     router.attach("/process-instance", ProcessInstanceResource.class);
+    router.attach("/process-instance/{processInstanceId}/variables", ProcessInstanceResource.class);
+    
     router.attach("/processInstance/{processInstanceId}/diagram", ProcessInstanceDiagramResource.class);
     router.attach("/process-definition/{processDefinitionId}/form", ProcessDefinitionFormResource.class);
     
