diff --git modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskResponse.java modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskResponse.java
index 3a7d8cf..1433e1c 100644
--- modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskResponse.java
+++ modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskResponse.java
@@ -37,6 +37,7 @@ public class TaskResponse {
   String parentTaskId;
   int priority;
   String processDefinitionId;
+  String processDefinitionName;
   String processInstanceId;
   String taskDefinitionKey;
   String formResourceKey;
@@ -161,6 +162,14 @@ public class TaskResponse {
     return processInstanceId;
   }
 
+  public String getProcessDefinitionName() {
+      return processDefinitionName;
+  }
+
+  public void setProcessDefinitionName(String processDefinitionName) {
+      this.processDefinitionName = processDefinitionName;
+  }
+
   public void setProcessInstanceId(String processInstanceId) {
     this.processInstanceId = processInstanceId;
   }
diff --git modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TasksResource.java modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TasksResource.java
index debb0a1..3b47049 100644
--- modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TasksResource.java
+++ modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TasksResource.java
@@ -14,11 +14,14 @@
 package org.activiti.rest.api.task;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import org.activiti.engine.ActivitiException;
+import org.activiti.engine.RepositoryService;
 import org.activiti.engine.impl.TaskQueryProperty;
 import org.activiti.engine.query.QueryProperty;
+import org.activiti.engine.repository.ProcessDefinition;
 import org.activiti.engine.task.TaskQuery;
 import org.activiti.rest.api.ActivitiUtil;
 import org.activiti.rest.api.DataResponse;
@@ -92,7 +95,21 @@ public class TasksResource extends SecuredResource {
       taskQuery.dueBefore(RequestUtil.parseToDate(strMaxDueDate));
     }
     
+    // add processDefinition name to every task
     DataResponse dataResponse = new TasksPaginateList().paginateList(getQuery(), taskQuery, "id", properties);
+    List<TaskResponse> tasks = (List<TaskResponse>) dataResponse.getData();
+    RepositoryService repositoryService = ActivitiUtil.getRepositoryService();
+    Map <String , ProcessDefinition> processDefinitions = new HashMap<String, ProcessDefinition>();
+    for (TaskResponse taskResponse : tasks) {
+       String processDefinitionId = taskResponse.getProcessDefinitionId();
+       ProcessDefinition processDefinition = processDefinitions.get(processDefinitionId);
+       if (processDefinition == null){
+           processDefinition = repositoryService.createProcessDefinitionQuery().processDefinitionId(processDefinitionId).list().get(0);
+           processDefinitions.put(processDefinitionId, processDefinition);
+       }
+       String processDefinitionName = processDefinition.getName();
+       taskResponse.setProcessDefinitionName(processDefinitionName);
+    }
     return dataResponse;
   }
 }
