diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/SecuredResource.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/SecuredResource.java
index 86f184d..69f9ee0 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/SecuredResource.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/SecuredResource.java
@@ -16,6 +16,7 @@ package org.activiti.rest.api;
 import java.util.List;
 
 import org.activiti.engine.identity.Group;
+import org.activiti.engine.impl.identity.Authentication;
 import org.activiti.rest.application.ActivitiRestApplication;
 import org.restlet.data.Status;
 import org.restlet.resource.ServerResource;
@@ -36,6 +37,7 @@ public class SecuredResource extends ServerResource {
   
   protected boolean authenticate(String group) {
     loggedInUser = ((ActivitiRestApplication) getApplication()).authenticate(getRequest(), getResponse());
+    Authentication.setAuthenticatedUserId(loggedInUser);
     if(loggedInUser == null) {
       // Not authenticated
       setStatus(Status.CLIENT_ERROR_FORBIDDEN, "Authentication is required");
diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskOperationResource.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskOperationResource.java
index 5476c27..89952ff 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskOperationResource.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/task/TaskOperationResource.java
@@ -57,6 +57,9 @@ public class TaskOperationResource extends SecuredResource {
       } else if ("complete".equals(operation)) {
         variables.remove("taskId");
         ActivitiUtil.getTaskService().complete(taskId, variables);
+      } else if ("reassign".equals(operation)){
+        String userId = properties.get("userId");
+        ActivitiUtil.getTaskService().setAssignee(taskId, userId);
       } else {
         throw new ActivitiException("'" + operation + "' is not a valid operation");
       }
