diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionResponse.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionResponse.java
index a440407..2c97dd1 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionResponse.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionResponse.java
@@ -33,6 +33,7 @@ public class ProcessDefinitionResponse implements Serializable {
   String diagramResourceName;
   String startFormResourceKey;
   boolean isGraphicNotationDefined;
+  String category;
 
   public ProcessDefinitionResponse(ProcessDefinitionEntity processDefinition) {
     this.setId(processDefinition.getId());
@@ -42,6 +43,7 @@ public class ProcessDefinitionResponse implements Serializable {
     this.setDeploymentId(processDefinition.getDeploymentId());
     this.setResourceName(processDefinition.getResourceName());
     this.setDiagramResourceName(processDefinition.getDiagramResourceName());
+    this.setCategory(processDefinition.getCategory());
   }
   
   public String getId() {
@@ -115,4 +117,13 @@ public class ProcessDefinitionResponse implements Serializable {
   public void setStartFormResourceKey(String startFormResourceKey) {
     this.startFormResourceKey = startFormResourceKey;
   }
+
+  public String getCategory() {
+    return category;
+  }
+
+  public void setCategory(String category) {
+    this.category = category;
+  }
+
 }
diff --git a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionsResource.java b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionsResource.java
index e45ab92..3aabfaf 100644
--- a/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionsResource.java
+++ b/modules/activiti-webapp-rest2/src/main/java/org/activiti/rest/api/process/ProcessDefinitionsResource.java
@@ -36,6 +36,7 @@ public class ProcessDefinitionsResource extends SecuredResource {
     properties.put("version", ProcessDefinitionQueryProperty.PROCESS_DEFINITION_VERSION);
     properties.put("deploymentId", ProcessDefinitionQueryProperty.DEPLOYMENT_ID);
     properties.put("name", ProcessDefinitionQueryProperty.PROCESS_DEFINITION_NAME);
+    properties.put("category", ProcessDefinitionQueryProperty.PROCESS_DEFINITION_CATEGORY);
   }
   
   @Get
